core/src/main/java/com/novoda/noplayer/model/PlayerVideoTrackCodecMapping.java

```diff
@@ -0,0 +1,30 @@
+package com.novoda.noplayer.model;
+
+import androidx.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class holds an association betwen the codec being used by a particular track
+ * and the decoder that the engine assigns to this codec.
+ *
+ * Bear in mind that this association is provisional and there is no guarantee that
+ * the engine will not fallback to another decoder. This is due to that usually several
+ * decoders are provided for a particual track. If the first decoder fails to initialise
+ * then the engine will choose another decoder. This information is not easely accessible
+ * and so we only provide the first association that the engine does.
+ */
+public final class PlayerVideoTrackCodecMapping {
```

Comment:
When is this cleared?

Reply:
I was thinking about that. Currently is not, it will be overridden by the next play. So that when new tracks comes in and new decoders are associated, the map is updated. It could be that the new stream has fewer tracks and then the map has invalid extra tracks, but if the request from the map comes from the actual tracks (that it is from the No-Player API point of view) then we should be fine. The information is attached to the VideoTrack.

Reply:
Ah I see. Yea this is not an issue then. 👍